### ISSUE TITLE
DataTable: when selecting one or multiple rows and performing any action on them selection length does not reset

### ISSUE THREAD

**Issue by JolianGof** [primefaces/primereact#5740]
### Describe the bug

i have a slot component that display the selection length and a delete button that delete the selected rows, when i perform the delete functionality and refetching the DataTable data the selection length does not reset 

### Reproducer

https://book-slight.codedamn.app/

### PrimeReact version

10.3.1

### React version

17.x

### Language

TypeScript

### Build / Runtime

Vite

### Steps to reproduce the behavior

1: go to codedamn url.
2: try to select one or multiple row.
3: try to delete the selected row/s.
4: it will delete the rows but the selection length would not change

### Expected behavior

i want the selection length to reset back to 0

**Comment by melloware** [comment]
Maybe its my connection but i can't get your reproducer to actually run.  It just hangs.

**Comment by JolianGof** [comment]
i was having a trouble with the reactQuery queryClient.invalidateQueries({queryKey:['users']}) if you try to open the reactQuery devtools and refetch the data after you delete a user you would see the issue im having

**Comment by JolianGof** [comment]
i don't know what was the problem with reproducer but i fixed it it should take you to the right link

**Comment by JolianGof** [comment]
sorry about that , but i added now the right reproducer, i hope to take in  consideration that i was having a problem with the reactQuery queryClient.invalidateQueries(['users]) if you try to open the reactQuery devtools and refetch the data after you delete a user you would see the problem i am having.

**Comment by melloware** [comment]
Ahhh your CodeDamn requires a login.  Any chance you could create a StackBlitz?

**Comment by JolianGof** [comment]
sorry of the delay, but i added Stackblitz project to the reproducer you can check it now

**Comment by melloware** [comment]
OK your example is pretty confusing.  Isn't your issue this `	selection={selection || table.selection}` you are telling it the selection is either your `useState` variable or if that is 0 then use the table.selection value?

**Comment by JolianGof** [comment]
ok i get the confusion, maybe i did not explain my issue right, my issue is that i have some actions that i perform to a row or multiple rows, such as adding , removing and deleting row/s, when i perform an action lets say adding to a row/s , note that i am using the checkbox to select one or multiple rows, when i perform the add action and then refresh the data table using the react query queryClient.invalidateQueries function the data table won't refresh and so when i select another row to perform that action the action happen to the rows i selected before and the newly selected row, the data table would only refresh when i make a page refresh. The user selected counter shows the selected row/s number and it also demonstrate what my issue is that the selection persist even after refreshing the data table. Hope the clarify my point

**Comment by melloware** [comment]
Yes that is correct because your control the 'selection' in a state variable. You need to clear that variable state right so the table thinks nothing is selected again?
